File: CursoPOOUber/Csharp/CCard.cs
using PaymentTypes;

namespace PaymentTypes.CCard
{

    //Definition of the "CCard" class. which inherits from "Payment" class.
    public class CCard : Payment
    {

        //Attributes
        public int Number { get; set; }
        public int CVV { get; set; }
        public DateTime Date { get; set; }

        public CCard(int id, int number, int cvv, DateTime date) : base(id)
        {
            this.Number = number;
            this.CVV = cvv;
            this.Date = date;
        }

        public void PrintCCardData()
        {
            Console.WriteLine("This is the Credit Card number: " + Number + " This is the secret code: " + CVV + " Payment was made on this date: " + Date);
        }
    }
}

File: CursoPOOUber/PHP/CCard.php
<?php
require_once('Payment.php');

//Definition of the "CCard" class, which inherits from "Payment" class.
class CCard extends Payment {

    //Attributes
    public $number;
    public $cvv;
    private $date;

    public function __construct($id, $number, $cvv, $date){
        parent::__construct($id);
        $this->number = $number;
        $this->cvv = $cvv;
        $this->date = new DateTime($date);
    }

    public function printCCardData() {
        echo "This is the Credit Card number: " + $this->number + " This is the secret cpde: " + $this->cvv + " Payment was made on this date: " + $this->date->format('2023-08-10');
    } 
}
?>